### PR TITLE
GH-39626: [Docs][R] Update NEWS.md for 15.0.0

### LINKED ISSUE #39626: [Docs][R] Update NEWS.md for 15.0.0
### Describe the enhancement requested

We need to update the news with the changes that were not included in 14.0.0.2 and 14.0.2.

### Component(s)

Documentation, R

### PR DESCRIPTION
**Pull request by assignUser**
### What changes are included in this PR?

Update NEWS.md with things not included in 14.0.2.

* Closes: #39626

### CODE REVIEW

**Review by thisisnic — CHANGES_REQUESTED** [review]
Thanks for updating this @assignUser! Would you mind [adding the GH usernames next to the PRs from folks who aren't regular contributors to the R package](https://style.tidyverse.org/news.html#acknowledgement)?

File: r/NEWS.md [hunk 1]
```diff
@@ -19,6 +19,17 @@
 
 # arrow 14.0.2.9000
 
+##  Minor improvements and fixes
+
+* Throw error when `string_replace` is passed vector of values in `pattern` (#39219)
```
Comment (thisisnic) [inline r/NEWS.md:24]:
```suggestion
* Throw error when the binding for `stringr::string_replace()` is passed vector of values in `pattern` (#39219)
```

Comment (nealrichardson) [reply r/NEWS.md:24]:
```
> stringr::string_replace
Error: 'string_replace' is not an exported object from 'namespace:stringr'
```

Isn't is `str_replace_all()`?

File: r/NEWS.md [hunk 2]
```diff
@@ -19,6 +19,17 @@
...
+* Document `open_dataset(format = "json")` (#38258)
+
+##  New features
```
Comment (nealrichardson) [inline r/NEWS.md:27]: Shouldn't New features go before Minor improvements and fixes?

File: r/NEWS.md [hunk 3]
```diff
@@ -19,6 +19,17 @@
...
+
+* Add `prod`-binding for `summarize()` (#38601)
```
Comment (nealrichardson) [inline r/NEWS.md:29]: As someone who hasn't been involved day-to-day in development, I have no idea what this means. Could this be revised for clarity?

File: r/NEWS.md [hunk 4]
```diff
@@ -19,6 +19,17 @@
 
 # arrow 14.0.2.9000
 
+##  Minor improvements and fixes
```
Comment (nealrichardson) [inline r/NEWS.md:22]: Seems like we should note the build changes we just merged under minor improvements

File: r/NEWS.md (hunk 1 above)
Comment (assignUser) [reply r/NEWS.md:24]: It seems it's for the family of `sub` type functions https://github.com/apache/arrow/pull/39219/files#diff-e992712511690afeafbd88380b92e2ffb2f06453f6b9122afda7273a44a4aca4R351

File: r/NEWS.md (hunk 3 above)
Comment (assignUser) [reply r/NEWS.md]: The title is a bit misleading it turns out, it just introduced bindings for `base::prod`

**Comment by amoeba** [comment]
Hey @assignUser @thisisnic, I went through the PR and (1) added some items that were missing and (2) made suggestions for existing items that I think improve each. @assignUser I made the suggestions easy to accept but I think the order they go in might be improved. Up to you.

I also did a quick review of all R-related commits in 15.0.0 (except those added in the last week or so) and think the list is now complete except for some mention of all the build system work (as @nealrichardson mentioned). Happy to review whatever item gets added for that.

**Pull request merged**